pkg/api/doctor/doctor.go

```diff
@@ -0,0 +1,51 @@
+package doctor
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/hospital-management/db"
+	"github.com/hospital-management/pkg/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func CreateDoctor(w http.ResponseWriter, r *http.Request) {
+	var doc utils.Doctor
+	err := json.NewDecoder(r.Body).Decode(&doc)
+	if err != nil {
+		log.Println("error occured during decoding input body", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if doc.Name == "" || doc.MobileNo == 0 {
+		log.Println("invalid credentials")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	collection := db.ConnectDB().Collection(utils.Doctors)
+	query := bson.M{"mobileno": doc.MobileNo}
+	findResult := collection.FindOne(context.TODO(), query)
+	var findData utils.Doctor
+	err = findResult.Decode(&findData)
+	if err != nil && err != mongo.ErrNoDocuments {
```

Comment:
There should be two checks after this:
1. If err == nil :: return saying data is already exists and return with status conflict.
2. if err != nil && err != mongo.ErrNoDocument: some db server error and return with internal error with error message in response.